archaius2-persisted2/src/main/java/com/netflix/archaius/persisted2/Persisted2ClientConfig.java

```diff
@@ -46,4 +46,12 @@
      * @return
      */
     String getServiceUrl();
+
+    /**
+     * When set to true, the initial loading of persistent properties has to be successful. Otherwise,
+     * the client will keep re-trying and Archaius initialization will not finish. In turn, the application's
+     * initialization will not finish and application will not start up. This is going to be the default
+     * behavior of persistent client.
+     */
+    boolean getSyncInit();
```

Comment:
Now that we're exposing this via configuration let's think of a better name.  How about something 'blockUntilInitialized'?